test/hotspot/jtreg/compiler/vectorization/TestFloat16ToFloatConv.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+* @test
+* @key randomness
+* @bug 8350835
+* @summary Test bug fix for JDK-8350835 discovered through Template Framework
+* @library /test/lib /
+* @run main/othervm compiler.vectorization.TestFloat16ToFloatConv
+*/
+
+package compiler.vectorization;
+
+import compiler.lib.ir_framework.*;
+import jdk.test.lib.Utils;
+import java.util.Random;
+
+public class TestFloat16ToFloatConv {
+    private static final Random RANDOM = Utils.getRandomInstance();
+    private static final int SIZE = 1024;
+    private static byte[] aB = new byte[SIZE];
+    private static char[] aC = new char[SIZE];
+    private static short[] aS = new short[SIZE];
+    private static int[] aI = new int[SIZE];
+    private static long[] aL = new long[SIZE];
+    private static float[] goldB, goldC, goldS, goldI, goldL;
+
+    static {
+        for (int i = 0; i < aB.length; i++) {
+            aB[i] = (byte)RANDOM.nextInt();
+            aC[i] = (char)RANDOM.nextInt();
+            aS[i] = (short)RANDOM.nextInt();
+            aI[i] = RANDOM.nextInt();
+            aL[i] = RANDOM.nextLong();
+        }
+        goldB = testByteKernel(aB);
+        goldC = testCharKernel(aC);
+        goldS = testShortKernel(aS);
+        goldI = testIntKernel(aI);
+        goldL = testLongKernel(aL);
+    }
+
+    @Test
+    // Not vectorized due to JDK-8350835
+    @IR(failOn = { IRNode.VECTOR_CAST_HF2F })
+    public static float[] testByteKernel(byte[] barr) {
+        float[] res = new float[barr.length];
+        for (int i = 0; i < barr.length; i++) {
+            res[i] = Float.float16ToFloat(barr[i]);
+        }
+        return res;
+    }
+
+    @Test
+    @IR(counts = {IRNode.VECTOR_CAST_HF2F, "> 0"},
+        applyIfOr = {"UseCompactObjectHeaders", "false", "AlignVector", "false"},
+        applyIfPlatformOr = {"x64", "true", "aarch64", "true", "riscv64", "true"},
```

Comment:
Can you justify the need to compressed object header usage,  it will mainly impact the pre-loop trip count compuation. AlignVector should be sufficient since it's a whitelisted option

Reply:
This check is taken from compiler/vectorization/TestFloatConversionsVector.java which also has float16 conversion tests to be in sync.

Reply:
If I remove UseCompressedHeaders check then the test will start failing for folks working on compressed headers so good to keep it there and as I mentioned before it is good to be in sync with other Float16ToFloat conversion test.